### ISSUE TITLE
parenting page filter box

### ISSUE THREAD

**Issue by sbanati** [topcowmoo/areca-vestiaria#13]
This filter box will be similar to the home page, but have its specific sub categories of

- Local Services
- ADHD
- Bullying
- Sex Discussion
- General

**Comment by topcowmoo** [comment]
https://github.com/topcowmoo/areca-vestiaria/pull/17

**Comment by sbanati** [comment]
some changes to think about : 

1. Filtering logic: 

So the way you handled filtering logic currently is ; 

filterData is defined with the useCallBack hook. It executes the filtering in 2 separate steps and then you log the filtered data and when the dependency changes the useEffect hook triggers filterData 

A more efficient way could be : 

filterdata is still defined in the useCallBack but COMBINE  the filtering steps so there is only 1 single pass and have an early exit 

then we dont need to have the useEffect hook, instead just call filterData directly in the component body 


TLDR ; we still use useCallback to memoize the filterData, we combine the filtering logic into a single pass , we remove the useEffect , all of this improves the performance by reducing the number of passes over the data and adding an early exit if no filters are active. 



I think this is your next step , since the code you provided is working and its good, consider it a brute force. Now you can clean it up and make it more efficient.

**Comment by topcowmoo** [comment]
https://github.com/topcowmoo/areca-vestiaria/pull/18

**Comment by topcowmoo** [comment]
![Image](https://github.com/user-attachments/assets/c9a431b8-e7bd-43f0-8cbe-0e8384940419)
